src/bridge/transactions/challenge.rs

```diff
@@ -117,7 +110,14 @@ impl ChallengeTransaction {
     ) {
         let input_index = 1;
 
-        self.tx.input[input_index].previous_output = input;
```

Comment:
This is another case where we should decide if we want to allow users to append multiple inputs. In this case, I think adding multiple inputs could make sense but I think we also need to handle line 123 too then and allow the value of each input to de user configurable.

We'd want a check somewhere to ensure the sum of the additional input values >= self.input_amount_crowdfunding;

Maybe finalize is a good place for this.

Reply:
For later... if users can specify an input we may also need to allow them to specify at least 1 output for refunds in the case they overpay with inputs. I think we could limit this to a single additional output.

Reply:
I refactored `add_input` function in `challenge` tx.
User can now append multiple inputs with distinct amounts. All amounts must sum up to >= `self.input_amount_crowdfunding`. If the amount is larger that expected, we add refund output to the tx.

Reply:
Do we have to be careful with users calling this multiple times? I think we'd better write it so it's idempotent (doesn't violate any constraints if they call it multiple times) or panic if they call it again by accident.